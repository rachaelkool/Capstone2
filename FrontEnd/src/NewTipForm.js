import React, { useState, useContext } from "react";
import UserContext from "./UserContext";
import DashboardApi from "./api/api";


function NewTipForm({addTip, date}) {
    const { currentEmployee } = useContext(UserContext);

    const [formData, setFormData] = useState({});

    async function handleSubmit(e) {
        e.preventDefault();
        let newTip;
        let tipData = {
            date: date,
            total_sales: formData.total_sales,
            total_tips: formData.total_tips,
            entered_by: currentEmployee.empId
        };
        addTip({...tipData, first_name: currentEmployee.firstName, last_name:currentEmployee.lastName })
        setFormData({})

        try {
            newTip = await DashboardApi.createTip(tipData);
        } catch (e) {
            return;
        }
        
        setFormData(data => ({ ...data }));
    }

    function handleChange(e) {
        const { name, value } = e.target;
        setFormData(data => ({ ...data, [name]: value }));
    }

    return (
        <div>
            <div className='form'>
                <form className="ui form" onSubmit={handleSubmit}>
                <h3>Input Tips</h3>
                <div className="field">
                    <label htmlFor="total_sales">Total Sales:</label>
                    <input
                        name="total_sales"
                        type="number"
                        min="0"
                        step="0.01"
                        value={formData.total_sales}
                        onChange={handleChange}
                    />
                </div>
                <div className="field">
                    <label htmlFor="total_tips">Total Tips:</label>
                    <input
                        name="total_tips"
                        type="number"
                        min="0"
                        step="0.01"
                        value={formData.total_tips}
                        onChange={handleChange}
                    />
                </div>
                <button>Add</button>
                </form>
            </div>
        </div>
    );
}


export default NewTipForm;