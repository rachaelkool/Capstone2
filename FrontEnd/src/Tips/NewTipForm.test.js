import React from "react";
import { render } from "@testing-library/react";
import NewTipForm from "./NewTipForm";


jest.mock('react', () => {
    const ActualReact = jest.requireActual('react')
    return {
      ...ActualReact,
      useContext: () => ({ empId: 101 }), 
    }
})

it("renders without crashing", function() {
    render(<NewTipForm />);
});
  
it("matches snapshot", function() {
    const { asFragment } = render(<NewTipForm />);
    expect(asFragment()).toMatchSnapshot();
});