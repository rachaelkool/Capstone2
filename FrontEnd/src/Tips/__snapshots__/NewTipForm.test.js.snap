// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form"
    >
      <form
        class="ui form"
      >
        <h3>
          Input Tips
        </h3>
        <div
          class="field"
        >
          <label
            for="total_sales"
          >
            Total Sales:
          </label>
          <input
            min="0"
            name="total_sales"
            step="0.01"
            type="number"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            for="total_tips"
          >
            Total Tips:
          </label>
          <input
            min="0"
            name="total_tips"
            step="0.01"
            type="number"
            value=""
          />
        </div>
        <button>
          Add
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
